### PR TITLE
Add the addFileNameHash option that allows users to remove the hash f…

### PR DESCRIPTION
**Pull request by GabrielLomba**
Add the addFileNameHash option that allows the removal of the hash from file names, as discussed in https://github.com/parse-community/parse-server/issues/4899.

### CODE REVIEW

**Comment by GabrielLomba** [comment]
@flovilmart Made the requested changes.

**Comment by flovilmart** [comment]
@GabrielLomba thanks for the changes, it's looking good. Can you revert the changes in the package-lock.json file please? I'M not sure they are relevant to this PR :0

**Comment by GabrielLomba** [comment]
@flovilmart Done.

**Comment by flovilmart** [comment]
Perfect thansk!

File: spec/FilesController.spec.js [hunk 1]
```diff
@@ -62,4 +62,40 @@ describe("FilesController",() =>{
...
+
+    filesController.createFile(config, fileName)
+
+    expect(gridStoreAdapter.createFile).toHaveBeenCalledTimes(1)
+    expect(gridStoreAdapter.createFile.calls.mostRecent().args[0]).toMatch(`^.{32}_${regexEscapedFileName}$`)
+
+    done();
+  });
+
+  it("should not add a unique hash to the file name when the preserveFileName option is false", (done) => {
```
[the remark this answers is not in the document]
Comment (GabrielLomba) [reply spec/FilesController.spec.js]: Oops. My mistake. Should I make another PR fixing it?

[the remark this answers is not in the document]
Comment (flovilmart) [reply spec/FilesController.spec.js]: Yep go ahead ;)

**Pull request merged**